src/animals/Eagle.ts

```diff
@@ -0,0 +1,41 @@
+ import { AnimalCandidate } from "../hierarchy/Animal"
+import Birds from "../hierarchy/Birds"
+import Reptile from "../hierarchy/Reptile"
+import { ICanFly } from "../interfaces/ICanFly"
+import { ICanTalk } from "../interfaces/ICanTalk"
+const EAGLE_SPACE_NEEDED_AS_ADULT = 100
+const EAGLE_SPACE_NEEDED_AS_CHILD = 50
+const NUMBER_OF_BABY_EAGLES_BORN = 4
+const AGE_TO_BE_ADULT = 1;
+export default class Eagle extends Birds implements ICanTalk, ICanFly {
+    private numberOfBabiesBorn = NUMBER_OF_BABY_EAGLES_BORN
+    protected home?: string | undefined
+    constructor(name: string, age: number, isMale: boolean) {
+        super(name, age, isMale)
+        this.isAdult = this.setMaturity(this.age,AGE_TO_BE_ADULT)
+        this.setHome('Animal Shelter')
```

Comment:
here you set the home to Animal Shelter but do you add the animal in the animal shelter anywhere?

Reply:
good point, we have to discuss it tomorrow